### PR TITLE
Initial commit of Cloud Search API

### PR DESCRIPTION
**Pull request by jgeewax**
This is an unlaunched API based on https://cloud.google.com/appengine/docs/python/search/documentclass. Docs aren't available yet, so this is pretty premature....

(The PM asked that I share this here despite all of that, working on getting everyone access to see docs and whatnot.)

### CODE REVIEW

File: docs/search-api.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Search
+------
```
Comment (dhermes) [inline docs/search-api.rst]: Is this file needed?

**Comment by dhermes** [comment]
@jgeewax Is this a `dont-merge` PR?

**Comment by jgeewax** [comment]
Yep - updated.

File: docs/search-api.rst (hunk 1 above)
Comment (jgeewax) [inline docs/search-api.rst]: Yea, it's the one that puts Search in the left side nav.

**Comment by dhermes** [comment]
Thanks.

File: docs/search-api.rst (hunk 1 above)
Comment (dhermes) [inline docs/search-api.rst]: Yeah but that's just a "trick" of CSS that we use for pages that actually exist.

It's moot since this isn't mean to be merged.

**Comment by jgeewax** [comment]
Closing this out and creating a new PR.
